### ISSUE TITLE
Abstraction in different layers responsibility is not good enough

### ISSUE THREAD

**Issue by sleepylee** [longngo1313/Nimbl3-Technical-Test#3]
- Such `MainScreenModel` shouldn't take that much responsibility to interact directly with SharePreferences, this may cause a tangled relationship when unit testing.

**Comment by longngo1313** [comment]
Yes. I should separate Data and Token to two different token. Thanks for your comment.
